CSharpFunctionalExtensions/Result/Dto/CSharpFunctionalExtensionsJsonSerializerOptions.cs

```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Text.Json;
+
+namespace CSharpFunctionalExtensions
```

Comment:
I'm sure we should not place cross-cutting functionality inside our root namespace.

Personally, I think that most users won't use the serialization features, and then they won't be happy to see intelligence polluted by types they don't need. 

I see that this particular type is internal, but my comment also applies to every type in the folder. 
And please don't put code inside namespaces you don't own, it makes it hard to understand the source of the methods and types used in the file (I'm basically talking about placing some code inside `System.*` namespace).

Reply:
Most libraries I saw that do things like that use their "main" namespace or the extended class for it.
For example extending ServiceCollection using the Microsoft.Extensions.DependencyInjection is a well know ASP.Core practice for discoverability. 
If a person uses Results near HttpResponseMessage, we want it those methods to be discoverable easilyt.